iam-login-service/src/main/java/it/infn/mw/iam/api/client/management/ClientManagementAPIController.java

```diff
@@ -140,6 +143,16 @@ public RegisteredClientDTO updateClient(@PathVariable String clientId,
     return managementService.updateClient(clientId, client);
   }
 
+  @PatchMapping("/{clientId}/status")
```

Comment:
I'd suggest two endpoints: `/{clientId}/enable` and `/{clientId}/disable` then the status is defined by the endpoint. The userId should be retrieved from the Authentication object in session|context with a
```
SecurityContextHolder.getContext().getAuthentication().getPrincipal()
```
But this means that if a client is accessing the API with a client credentials grant, the principal will be the client uuid. So you could check that clientId is equal to the principal before searching it into users repository (if needed).

Reply:
Also AccountUtils object can be used (it's a bean). Its method `getAuthenticatedUserAccount` can be used as seen in `AupSignaturePageController`

Reply:
Using AccountUtils. Do I need any additional check?

Reply:
I don't think. I'll resolve this conversation 👍